test/integration/infra.go

```diff
@@ -177,3 +184,42 @@ func (infra *infra) teardown() {
 func (infra *infra) kubeApply(yaml string) error {
 	return util.RunInput(fmt.Sprintf("kubectl apply -n %s -f -", infra.Namespace), yaml)
 }
+
+type response struct {
+	id      []string
+	version []string
+	port    []string
+	code    []string
+}
+
+func (infra *infra) clientRequest(app, url string, count int, extra string) response {
+	request, err := util.Shell(fmt.Sprintf("kubectl exec %s -n %s -c app -- client -url %s -count %d %s",
+		infra.apps[app][0], infra.Namespace, url, count, extra))
+	out := response{}
+	if err != nil {
+		glog.Errorf("client request error %v for %s in %s", err, url, app)
+		return out
+	}
+
+	ids := regexp.MustCompile("(?i)X-Request-Id=(.*)").FindAllStringSubmatch(request, -1)
```

Comment:
clientRequest is called many times from nested loops. It would be good to move MustCompile out to a global to avoid recompiling regex each time.